### ISSUE TITLE
Add parameter binding test

### ISSUE THREAD

**Issue by greenlaw110** [TechEmpower/FrameworkBenchmarks#2842]
Many full stack frameworks provides parameter bindings. 

For example suppose there are several application defined classes (in Java)

```java
public class Foo {
    public String id;
    public int value;
}
```

```java
public class Bar {
    public Foo foo;
    public boolean flag;
}
```

Application controller that needs to create `Bar` instance:

```java
@PostAction("/bars")
public Bar create(Bar bar) {
    return bar;
}
```

The parameter binding could happen in two difference scenarios:

1. `Content-Type` is `application/x-www-form-urlencoded`

either

```
bar[foo][id]=x
bar[foo][value]=1
bar[flag]=true
```

or 

```
bar.foo.[id]=x
bar.foo.value=1
bar.flag.=true
```

2. `Content-Type` is `application/json`

either:

```json
{
    "foo": {"id": "x", "value": 1},
    "flag": true
}
```

or 

```json
{
    "bar" : {
        "foo": {"id": "x", "value": 1},
        "flag": true
    }
}
```

When framework received the `POST` request along with property set `Content-Type` header and post data, it shall resolve the data and bind to the handler parameter `Bar bar` in the function.

**Comment by nbrady-techempower** [comment]
There might be a place for this as a piece of the routing test we're working on

**Comment by greenlaw110** [comment]
Great. But the parameter binding test and routing test are two very different thing. One is testing how framework handles large routing table (might including URL path variable and even regex validation); The other one is about how framework preprocess the request parameters/form fields. 

I think every framework/platform can participate the routing test, but not all frameworks provides the parameter binding feature.

**Comment by nbrady-techempower** [comment]
> but not all frameworks provides the parameter binding feature.

Right, this is the part I was concerned with. Which wouldn't make it a high priority for test types with out resources. Still seems like it could be worthwhile down the road.

**Comment by greenlaw110** [comment]
My understanding is there is always feature zones that only a part of framework/platforms can attend, e.g. not all framework platform support URL path variable, and maybe less supports regex matching for URL path variables.

TechEmpower just need to create the tech spec and leave the implementation to framework authors or other contributors...

**Comment by nbrady-techempower** [comment]
Right, your understanding is correct. However, the 6 tests (now 7) that we have probably could implemented by all the frameworks. There's actually an open issue (outdated) for the missing tests from the frameworks. 

Not saying it's not worthwhile

**Comment by NateBrady23** [comment]
Closing as part of bulk cleanup.
